web/static/js/menu.js

```diff
@@ -45,6 +45,21 @@ loadRelationForm.addEventListener('submit', e => {
 
     const relationIdInput = loadRelationForm.querySelector('input[name=relation-id]')
     relationId = parseInt(relationIdInput.value)
+    if (isNaN(relationId)) {
+        const osmOrgRelationRegex = /https:\/\/(?:www\.)?openstreetmap.org\/relation\/(\d+)/;
+        const reMatch = relationIdInput.value.match(osmOrgRelationRegex);
+        if (reMatch !== null) {
+            relationId = parseInt(reMatch[1]);
+        } else {
+            console.log("invalid relation");
+            showMessage(
+                'danger',
+                '❌ invalid relation ID or URL',
+                'The relation id or URL entered is invalid',
+            )
+            return;
+        }
+    }
```

Comment:
I believe a much simpler and more robust solution would be to ignore all non-numeric characters (0-9). This would enhance compatibility even further and should work perfectly fine. If we choose to adopt this approach, the dot (.) should also be ignored.

e.g.:

```
123 -> 123
123.123 -> 123123 or 123
https://example.com/123123/test?something -> 123123
```

However, there should be special handling for URL fragments (`#map=19/51.60259/19.55114`) as they may also contain numbers. In my opinion, parsing of the input should stop if a number has already been found and there is a non-numeric separator. So, the revised approach would be as follows:

```
123.456 -> 123 (stops after ., non-numeric char)
https://example.com/123123/test?something -> 123123
https://example.com/123123/test#something=456 -> 123123
https://example.com/test#something=456 -> 456
```

Finally, the parsing result should be reflected back to the user *immediately* by updating the input element value. Maybe listen on some onchange event. So if user tries to type some non-numeric characters, they are automatically removed.